File: api-chi/cmd/routes/movie_depend_movie_category_test.go
package routes

import (
	"api-chi/cmd/models"
	"api-chi/internal/message"

	"bytes"
	"encoding/json"
	"net/http"
	"net/http/httptest"
	"testing"

	"github.com/go-chi/chi/v5"
	"github.com/stretchr/testify/assert"
)

func Test_MovieDependMovieCategoryRoutes(t *testing.T) {
	r := chi.NewRouter()
	MovieDependMovieCategoryRoutes(r)
	id := ""

	t.Run("Count success", func(t *testing.T) {
		req := httptest.NewRequest("GET", "/movie-depend-movie-category/count", nil)
		res := httptest.NewRecorder()

		r.ServeHTTP(res, req)

		assert.Equal(t, http.StatusOK, res.Code)
		var response message.Response
		err := json.NewDecoder(res.Body).Decode(&response)
		assert.NoError(t, err)
		assert.Equal(t, message.GET_DATA_SUCCESS, response.Message)
		assert.NotNil(t, response.Data)
	})

	t.Run("Create success", func(t *testing.T) {
		input := models.MovieDependMovieCategory{
			MovieCategoryId: "360d60a6-3f3a-4c00-927a-04aa35410acc",
			MovieId:         "0bc66ff1-15ec-4d54-8830-1baf807b8091",
		}
		body, _ := json.Marshal(input)

		req := httptest.NewRequest("POST", "/movie-depend-movie-category", bytes.NewBuffer(body))
		res := httptest.NewRecorder()

		r.ServeHTTP(res, req)

		assert.Equal(t, http.StatusOK, res.Code)
		var response message.Response
		err := json.NewDecoder(res.Body).Decode(&response)
		assert.NoError(t, err)
		assert.Equal(t, message.CREATE_DATA_SUCCESS, response.Message)
		assert.NotNil(t, response.Data)

		idValue, ok := response.Data.(string)
		if !ok {
			t.Fatalf("Expected response.Data to be of type string, got %T", response.Data)
		}
		id = idValue
	})

	t.Run("Remove success", func(t *testing.T) {
		if id == "" {
			t.Fatal("ID must be set before running Remove test")
		}

		req := httptest.NewRequest("DELETE", "/movie-depend-movie-category/"+id, nil)
		res := httptest.NewRecorder()

		r.ServeHTTP(res, req)

		assert.Equal(t, http.StatusOK, res.Code)
		var response message.Response
		err := json.NewDecoder(res.Body).Decode(&response)
		assert.NoError(t, err)
		assert.Equal(t, message.REMOVE_DATA_SUCCESS, response.Message)
	})
}

File: api-fastify/src/tests/routes/movie_depend_movie_category.test.ts
import fastify from "fastify"
import MovieDependMovieCategoryRoutes from "@/api/routes/movie_depend_movie_category"
import { MovieDependMovieCategory } from "@/types/movie_depend_movie_category"
import { HttpStatus } from "@/types/http_status"

describe("MovieCategoryRoutes", () => {
  let app: ReturnType<typeof fastify>
  let id: string = ""

  beforeAll(async () => {
    app = fastify()
    app.register(MovieDependMovieCategoryRoutes)
    await app.ready()
  })

  afterAll(async () => {
    await app.close()
  })

  describe("POST /movie-depend-movie-categories", () => {
    it("should create a movie depend movie category", async () => {
      const input: MovieDependMovieCategory = {
        id: "",
        movieCategoryId: "360d60a6-3f3a-4c00-927a-04aa35410acc",
        movieId: "0bc66ff1-15ec-4d54-8830-1baf807b8091",
      }

      const response = await app.inject({
        method: "POST",
        url: "/movie-depend-movie-categories",
        payload: input,
      })

      expect(response.statusCode).toBe(HttpStatus.OK)
      const result = response.json()
      expect(result.success).toBe(true)
      expect(typeof result.data.id).toBe("string")
      expect(typeof result.data.movieCategoryId).toBe("string")
      expect(typeof result.data.movieId).toBe("string")
      expect(result.data.movieCategoryId).toBe(input.movieCategoryId)
      expect(result.data.movieId).toBe(input.movieId)
      id = result.data.id
    })
  })

  describe("DELETE /movie-depend-movie-categories/:id", () => {
    it("should remove a movie depend movie category", async () => {
      const response = await app.inject({
        method: "DELETE",
        url: `/movie-depend-movie-categories/${id}`,
      })

      expect(response.statusCode).toBe(HttpStatus.OK)
      const result = response.json()
      expect(result.success).toBe(true)
    })
  })
})
